lib/web_ui/lib/src/engine/vector_math.dart

```diff
@@ -1091,315 +1062,21 @@ class Matrix4 {
   }
 }
 
-/// 3D column vector.
-class Vector3 {
-  /// Construct a new vector with the specified values.
-  factory Vector3(double x, double y, double z) =>
-      Vector3.zero()..setValues(x, y, z);
-
-  /// Zero vector.
-  Vector3.zero() : _v3storage = Float32List(3);
-
-  /// Splat [value] into all lanes of the vector.
-  factory Vector3.all(double value) => Vector3.zero()..splat(value);
-
-  /// Copy of [other].
-  factory Vector3.copy(Vector3 other) => Vector3.zero()..setFrom(other);
-
-  /// Constructs Vector3 with given Float32List as [storage].
-  Vector3.fromFloat32List(this._v3storage);
-
-  /// Constructs Vector3 with a [storage] that views given [buffer] starting at
-  /// [offset]. [offset] has to be multiple of [Float32List.bytesPerElement].
-  Vector3.fromBuffer(ByteBuffer buffer, int offset)
-      : _v3storage = Float32List.view(buffer, offset, 3);
-
-  /// Generate random vector in the range (0, 0, 0) to (1, 1, 1). You can
-  /// optionally pass your own random number generator.
-  factory Vector3.random([math.Random? rng]) {
-    rng ??= math.Random();
-    return Vector3(rng.nextDouble(), rng.nextDouble(), rng.nextDouble());
-  }
-
-  final Float32List _v3storage;
+const Vector3 kUnitX = (x: 1.0, y: 0.0, z: 0.0);
+const Vector3 kUnitY = (x: 0.0, y: 1.0, z: 0.0);
+const Vector3 kUnitZ = (x: 0.0, y: 0.0, z: 1.0);
```

Comment:
Nice! This actually guarantees immutability, unlike the previous method of leaving comments with warnings.